path/utils.go

```diff
@@ -0,0 +1,32 @@
+package path
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// exist checks file or directory existence
+func exist(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	return os.IsExist(err)
```

Comment:
Use `errors.Is(err, fs.ErrExist)` as suggested in https://pkg.go.dev/os#IsExist

Reply:
Sure. Use `errors.Is(err, fs.ErrExist)` in unionDir.go